saml-core/src/main/java/org/keycloak/saml/processing/core/saml/v2/writers/BaseWriter.java

```diff
@@ -174,6 +175,8 @@ public void writeAttributeTypeWithoutRootTag(AttributeType attributeType) throws
                     	writeNameIDTypeAttributeValue((NameIDType) attributeValue);
                     } else if (attributeValue instanceof XMLGregorianCalendar) {
                         writeDateAttributeValue((XMLGregorianCalendar) attributeValue);
+                    } else if (attributeValue instanceof Element) {
+                        StaxUtil.writeDOMElement(writer, (Element) attributeValue);
```

Comment:
Currently it is expected that `attributeValue` starts with `AttributeValue` element. Yet for the sake of consistency with the rest of the values and enforcing schema compliance, the attribute value should only be the value without `AttributeValue`. In other words, introduce a `writeElementAttributeValue(Element attributeValue)` method similar to e.g. `writeNameIDTypeAttributeValue`

Reply:
Thanks for the suggestions - I changed the code in the proposed way. Because the type of the AttributeValue is variable, I created a new AttributeValueElementType class to contain both the type and the element.